src/Microsoft.Azure.WebJobs.Host/Triggers/TriggerArgumentBinding/SimpleTriggerArgumentBinding.cs

```diff
@@ -125,5 +129,30 @@ protected void AddToBindingData(IBindingDataProvider provider, Dictionary<string
                 }
             }
         }
+
+        private class ReturnValueProvider : IValueProvider
+        {
+            public Type Type => typeof(RetryResult);
+
+            public object Value;
+
+            public Task<object> GetValueAsync()
+            {
+                return Task.FromResult(Value);
+            }
+            public string ToInvokeString()
+            {
+                return Value.ToString();
+            }
+        }
+
+        private class ReturnValueBinder : ReturnValueProvider, IValueBinder
```

Comment:
Why is this defined as a separate class? Can we just implement `IValueBinder` above?